### ISSUE TITLE
Support verbatim fields

### ISSUE THREAD

**Issue by mutlusun** [typst/biblatex#21]
Hello,

Many thanks for your work on the `biblatex` crate. I use it in a little converstion tool: https://codeberg.org/mutluyuz/csv2bibtex

When working with the crate, I noticed that it does not support verbatim fields right now (like url or file). See this discussion on stackexchange: https://tex.stackexchange.com/questions/177201/bibtex-biblatex-verbatim-fields-and-special-characters

Would it be possible to support such unescaped fields as well (in their verbatim mode)?

Thanks for your help!

**Comment by reknih** [comment]
Good call!

I implemented this such that only braces, backslashes and colons (Zotero file output; don't ask me) must be escaped in verbatim fields. The feature is in the 0.6 release (on crates.io in a few minutes).

**Comment by mutlusun** [comment]
Thanks for your fast reply and implementation!

Sorry for my late reply, I had to test some things with this new feature. Can you explain, why colons are also escaped? For example, the url `https://www.example.com` gets `https\://www.example.com`. This way, some bibliography managers like JabRef does not detect/open this url in their GUI.

Best

**Comment by reknih** [comment]
I escaped this because Zotero escapes the colon in paths on Windows when exporting (at least with the Better BibTeX plugin).
I couldn't find a general specification for which characters need to be escapable in verbatim mode anywhere (for example, the braces must stay escapable) so I opted for compatibility. What do you think about this? Is this a bug with Zotero or do you have access to a more formal spec?

I hope the new implementation is otherwise useful to you!

**Comment by mutlusun** [comment]
Thanks for your answer. Indeed, the new implementation is useful to me! This way, I can parse CSV files and leave the URL fields untouched.

Referring to the [biblatex documentation](https://ftp.rrze.uni-erlangen.de/ctan/macros/latex/contrib/biblatex/doc/biblatex.pdf), it seems that there is no clear definition of the verbatim mode (at least when searching for `verbatim`). However, the following parts might be relevant:

> Verbatim fields are processed in verbatim mode and may contain special characters.

This sounds like all special characters are left as is.

> Since the 〈resource〉 string is read in a verbatim-like mode, it may contain arbitrary characters. The only restriction is that any curly braces must be balanced (p. 90)

This sounds like only curly braces must be escaped / balanced (makes sense 😄  ).

> Note that eprint is a verbatim field. Always give the identifier in its unmodified form. For example, there is no need to replace _ with \_. (p. 149)

Same as above.

> urlraw field (verbatim)
>
> This is the unencoded, raw version of any url. This is intended for use when the display version and clickable link version of a URL are different. This can be the case when the URL contains special or Unicode characters. In the case where no such characters occur, may be identical to the url.

This sounds like `url` might be escaped for some special characters. However, I don't understand what this could be and have never seen it.

To summarize, I agree with your decision ;) But this might be interesting for later as a reference. I will see, whether I will find the time this week and open a PR.
